File: librarian-app/src/main/java/com/github/dkurata38/liblio/librarian_app/security/LibrarianUser.java
package com.github.dkurata38.liblio.librarian_app.security;

import lombok.Getter;

import java.util.Collection;
import org.springframework.security.core.GrantedAuthority;
import org.springframework.security.core.userdetails.User;

public class LibrarianUser extends User {
    @Getter
    private final String librarianId;

    @Getter
    private final String libraryId;

    public LibrarianUser(String librarianId, String libraryId, String username, String password, Collection<? extends GrantedAuthority> authorities) {
        super(username, password, authorities);
        this.librarianId = librarianId;
        this.libraryId = libraryId;
    }

    public LibrarianUser(String librarianId, String libraryId, String username, String password, boolean enabled, boolean accountNonExpired, boolean credentialsNonExpired, boolean accountNonLocked, Collection<? extends GrantedAuthority> authorities) {
        super(username, password, enabled, accountNonExpired, credentialsNonExpired, accountNonLocked, authorities);
        this.librarianId = librarianId;
        this.libraryId = libraryId;
    }
}

File: librarian-app/src/main/kotlin/com/github/dkurata38/liblio/librarian_app/security/LibrarianUser.kt
package com.github.dkurata38.liblio.librarian_app.security

import org.springframework.security.core.GrantedAuthority
import org.springframework.security.core.userdetails.User

class LibrarianUser(
    val librarianId: String,
    val libraryId: String,
    username: String,
    password: String,
    enabled: Boolean,
    accountNonExpired: Boolean,
    credentialsNonExpired: Boolean,
    accountNonLocked: Boolean,
    authorities: MutableCollection<out GrantedAuthority>
) : User(username, password, enabled, accountNonExpired, credentialsNonExpired, accountNonLocked, authorities) {
    constructor(librarianId: String, libraryId: String, username: String,
                password: String, authorities: MutableCollection<out GrantedAuthority>): this(librarianId, libraryId, username, password, true, true, true, true, authorities)
}
